compiler/lib/ocaml_compiler.ml

```diff
@@ -18,11 +18,15 @@
 
 open! Stdlib
 
-let rec constant_of_const : _ -> Code.constant =
+let rec constant_of_const c : Code.constant =
   let open Lambda in
   let open Asttypes in
-  function
-  | Const_base (Const_int i) -> Int (Int32.of_int_warning_on_overflow i)
+  match c with
+  | Const_base (Const_int i) ->
+      Int
+        (match Config.target () with
+        | `JavaScript -> Int32.of_int_warning_on_overflow i
+        | `Wasm -> Int31.(of_int_warning_on_overflow i |> to_int32))
```

Comment:
`Const_base (Const_int32 i) -> Int i` and `| Const_base (Const_nativeint i) ` below need to be updated.

Reply:
True. Done